test_swerve_controller:
  ros__parameters:
    use_stamped_vel: false

    drive_joints_names: [ "fl_drive_joint", "fr_drive_joint", "rl_drive_joint", "rr_drive_joint" ]
    steer_joints_names: [ "fl_steer_joint", "fr_steer_joint", "rl_steer_joint", "rr_steer_joint" ]

    drive_joints_state_names: [ "fl_drive_joint", "fr_drive_joint", "rl_drive_joint", "rr_drive_joint" ]
    steer_joints_state_names: [ "fl_steer_joint", "fr_steer_joint", "rl_steer_joint", "rr_steer_joint" ]

    wheel_radius: 0.2  # m
    wheelbase: 1.8
    wheel_track: 2.3
    drive_to_steer_offset: 0.0

    publish_rate: 50.0
    odom_frame_id: odom
    base_frame_id: base_link
    pose_covariance_diagonal : [0.001, 0.001, 0.001, 0.001, 0.001, 0.01]
    twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.01]

    open_loop: false
    enable_odom_tf: true
    position_feedback: false
    cmd_vel_timeout: 0.5
    publish_limited_velocity: true
    velocity_rolling_window_size: 10

    # Velocity and acceleration limits
    # Whenever a min_* is unspecified, default to -max_*
    linear.x.has_velocity_limits: true
    linear.x.has_acceleration_limits: true
    linear.x.has_jerk_limits: false
    linear.x.max_velocity: 1.0
    linear.x.min_velocity: -1.0
    linear.x.max_acceleration: 1.0
    linear.x.max_jerk: 0.0
    linear.x.min_jerk: 0.0

    linear.y.has_velocity_limits: true
    linear.y.has_acceleration_limits: true
    linear.y.has_jerk_limits: false
    linear.y.max_velocity: 1.0
    linear.y.min_velocity: -1.0
    linear.y.max_acceleration: 1.0
    linear.y.max_jerk: 0.0
    linear.y.min_jerk: 0.0

    angular.z.has_velocity_limits: true
    angular.z.has_acceleration_limits: true
    angular.z.has_jerk_limits: false
    angular.z.max_velocity: 1.0
    angular.z.min_velocity: -1.0
    angular.z.max_acceleration: 1.0
    angular.z.min_acceleration: -1.0
    angular.z.max_jerk: 0.0
    angular.z.min_jerk: 0.0

